packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -421,8 +421,8 @@ class HtmlElementView extends StatelessWidget {
   /// used by the app to customize the element by adding attributes and styles.
   ///
   /// ```dart
+  /// import 'package:flutter/src/web.dart' as web;
```

Comment:
Not sure if this is okay :)

Reply:
What was wrong with the old code?

Reply:
Dartdoc/analysis would attempt to run this code, but because `package:web` is no longer a dependency, it would not resolve.

Reply:
Sorry for derailing the PR, but I'm starting to have second thoughts about it:

This example is meant to work when developers copy it into their apps, but it looks like it won't? This might be an indicator that removing the dependency on `package:web` may not be a good idea. If users are expected to use the core framework APIs along with `package:web` then for all intents and purposes the framework already depends on it despite the hacks we use to sever the dependency for `pub get`'s sake.

Maybe we should pay the cost of pinning so our developers do not pay the cost of poor ergonomics. In fact, now that I think about this class, it is a web-specific class but it lives in the platform-agnostic `widgets/platform_view.dart` library. We should move it into its own `widgets/platform_view_web.dart` library and teach it to speak in terms of `package:web` interfaces instead of `Object`. That library will be imported behind a conditionally imported library that has access to web-specific libraries, including `package:web`.

Reply:
For EXAMPLES, folks SHOULD use `pkg:web`.

Casting works fine. Types are erased.
@srujzs is just working around tooling silly here.

That's the beauty here. Let's lean into it!

Reply:
It should still work. This is an analysis issue that is specific to the doc comment. Users should still import `package:web/web.dart` and the code will still work fine even with the shim we introduce here. `@staticInterop` (and later, extension types on `JSObject`) can be casted freely from one type to another unlike `dart:html` types, because we only ever check that the type is a `JSObject`. I don't want people to be able to use the shim besides for the purpose in this CL.

If we want to treat this comment as a guide for users (which I agree we should), then it's a matter of ignoring that analysis issue.

Reply:
How about we move the example to flutter.dev and remove it from the docs?

Reply:
SGTM! At a minimum just open the issue. This can go in our embedding docs

Reply:
I've removed the example from this PR and filed https://github.com/flutter/flutter/issues/140010.